### PR TITLE
Optimized the partitioning strategy implementation details to avoid unnecessarily high RU usage

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description
This PR optimizes some of the partitioning strategy values to reduce chance that RU-consumption for queries or change feed requests is unnecessarily high when no EPK filtering is enabled for an account. Functionally using Spark partitions scoped to subsets of physical partitions is possible to increase parallelization but it comes with an overhead of increase RU usage - and in many cases that is unnecessary. So, this PR avoids this from happening in most cases (unless partition merge happened) for all partitioning strategies except for Aggressive and Custom.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos-spark_3_2-12/Samples/Scala/NYC-Taxi-Data/01_Batch.scala [hunk 1]
```diff
@@ -255,7 +255,7 @@ val readCfg = Map(
   "spark.cosmos.accountKey" -> cosmosMasterKey,
   "spark.cosmos.database" -> "SampleDatabase",
   "spark.cosmos.container" -> "GreenTaxiRecords",
-  "spark.cosmos.read.partitioning.strategy" -> "Default",
+  "spark.cosmos.read.partitioning.strategy" -> "Restrictive",
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/Samples/Scala/NYC-Taxi-Data/01_Batch.scala:258]: will this cause regressions for customers who is using 'Default' for the high concurrency?

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/Samples/Scala/NYC-Taxi-Data/01_Batch.scala:258]: Technically possible - but extremely unlikely given the high RU charges.

**Review by kushagraThapar — APPROVED** [review]
Changes look good, however, I am wondering if this is a breaking change? And should we explicitly call it out in the changelog instead of calling it out as `optimized`?

**Pull request merged**